sem_covid/adapters/abstract_store.py

```diff
@@ -1,85 +1,206 @@
+#!/usr/bin/python3
+
+# abstract_store.py
+# Date:  01/07/2021
+# Author: Stratulat Ștefan
+
+"""
+    This module aims to define interfaces for different types of stores that are used in the project.
+"""
+
 from abc import ABC, abstractmethod
 import pandas as pd
 import pathlib
 
 
 class ObjectStoreABC(ABC):
+    """
+        This class defines the abstraction for an object store, based on S3 technology.
+    """
 
     @abstractmethod
     def empty_bucket(self, object_name_prefix: str = None):
+        """
+            This method deletes all stored objects that begin with a specific prefix.
+        :param object_name_prefix: the prefix that will be searched in the name of the stored objects.
+        :return:
+        """
         raise NotImplementedError
 
     @abstractmethod
     def put_object(self, object_name: str, content) -> int:
+        """
+            This method allows you to store an object with a name so that it can be found later.
+        :param object_name: the name of the object to be stored
+        :param content: the content of the object to be stored
+        :return:
+        """
         raise NotImplementedError
 
     @abstractmethod
     def get_object(self, object_name: str):
+        """
+            This method allows you to extract an object based on its name.
+        :param object_name: the name of the desired object to be extracted
+        :return: returns the requested object based on the transmitted object_name
+        """
         raise NotImplementedError
 
     @abstractmethod
     def list_objects(self, object_name_prefix: str):
+        """
+            This method returns a list of objects that have been stored
+             and whose names begin with the prefix transmitted.
+        :param object_name_prefix: the prefix of the name based on which the objects will be searched
+        :return: returns a list of objects beginning with the prefix transmitted by object_name_prefix
```

Comment:
Please note that the returned list is not the list of actual objects, but some `minio.datatype.object` entities that contain only the metadata about the stored objects, not the content itself. I have created a fakeDatasToreObject for some tests to work.